lib/Backend/arm64/LowerMD.cpp

```diff
@@ -6008,73 +5958,71 @@ LowererMD::GenerateClz(IR::Instr * instr)
-    IR::Opnd * symDblDst;
-    IR::Opnd * symTypeDst;
-    IR::Instr *newInstr;
-    IR::Instr * numberInitInsertInstr = nullptr;
-    if (instrOrig->dstIsTempNumber)
-    {
-        // Use the original dst to get the temp number sym
-        StackSym * tempNumberSym = this->m_lowerer->GetTempNumberSym(instrOrig->GetDst(), instrOrig->dstIsTempNumberTransferred);
-
-        // LEA dst, &tempSym
-        IR::SymOpnd * symTempSrc = IR::SymOpnd::New(tempNumberSym, TyMachPtr, this->m_func);
-        newInstr = IR::Instr::New(Js::OpCode::LEA, dstOpnd, symTempSrc, this->m_func);
-        instrInsert->InsertBefore(newInstr);
-        LegalizeMD::LegalizeInstr(newInstr, false);
 
-        symVTableDst = IR::SymOpnd::New(tempNumberSym, TyMachPtr, this->m_func);
-        symDblDst = IR::SymOpnd::New(tempNumberSym, (uint32)Js::JavascriptNumber::GetValueOffset(), TyFloat64, this->m_func);
-        symTypeDst = IR::SymOpnd::New(tempNumberSym, (uint32)Js::JavascriptNumber::GetOffsetOfType(), TyMachPtr, this->m_func);
+    // s1 = MOV opndFloat
+    IR::RegOpnd *s1 = IR::RegOpnd::New(TyMachReg, m_func);
+    IR::Instr *mov = IR::Instr::New(Js::OpCode::FMOV_GEN, s1, opndFloat, m_func);
+    instrInsert->InsertBefore(mov);
 
-        if (this->m_lowerer->outerMostLoopLabel == nullptr)
-        {
-            // If we are not in loop, just insert in place
-            numberInitInsertInstr = instrInsert;
-        }
-        else
-        {
-            // Otherwise, initialize in the outer most loop top if we haven't initialize it yet.
-            numberInitInsertInstr = this->m_lowerer->initializedTempSym->TestAndSet(tempNumberSym->m_id) ?
-                nullptr : this->m_lowerer->outerMostLoopLabel;
-        }
-    }
-    else
+    if (m_func->GetJITFunctionBody()->IsAsmJsMode())
     {
-        this->GenerateNumberAllocation(dstOpnd, instrInsert, isHelper);
-        symVTableDst = IR::IndirOpnd::New(dstOpnd, 0, TyMachPtr, this->m_func);
-        symDblDst = IR::IndirOpnd::New(dstOpnd, (uint32)Js::JavascriptNumber::GetValueOffset(), TyFloat64, this->m_func);
-        symTypeDst = IR::IndirOpnd::New(dstOpnd, (uint32)Js::JavascriptNumber::GetOffsetOfType(), TyMachPtr, this->m_func);
-        numberInitInsertInstr = instrInsert;
-    }
+        // s1 = FMOV_GEN src
+        // tmp = UBFX s1, #52, #11 ; extract exponent, bits 52-62
+        // cmp = tmp, 0x7FF
+        // beq tmp, helper
+        // b done
+        // helper:
+        // tmp2 = tmp2 = UBFX s1, #0, #52 ; extract mantissa, bits 0-51
+        // cbz tmp2, done
+        // s1 = JavascriptNumber::k_Nan
+        // done:
 
-    if (numberInitInsertInstr)
-    {
-        IR::Opnd *jsNumberVTable = m_lowerer->LoadVTableValueOpnd(numberInitInsertInstr, VTableValue::VtableJavascriptNumber);
+        IR::RegOpnd* tmp = IR::RegOpnd::New(TyMachReg, m_func);
+
+        IR::Instr* newInstr = IR::Instr::New(Js::OpCode::UBFX, tmp, s1, IR::IntConstOpnd::New(((52<<16)|11), TyMachReg, m_func, true), m_func);
```

Comment:
It just landed, but there is a BITFIELD() macro you can use to say BITFIELD(52, 11) instead of hand-assembling the constant here.